src/noise.h

```diff
@@ -44,23 +44,23 @@ class PseudoRandom {
 public:
 	const static u32 RANDOM_RANGE = 32767;
 
-	inline PseudoRandom(int seed=0):
-		m_next(seed)
+	inline PseudoRandom(s32 seed_=0)
 	{
+		seed(seed_);
 	}
 
-	inline void seed(int seed)
+	inline void seed(s32 seed)
 	{
 		m_next = seed;
 	}
 
-	inline int next()
+	inline u32 next()
 	{
 		m_next = m_next * 1103515245 + 12345;
```

Comment:
Out of scope for this PR to fix, but isn't there some integer overflow UB here?

Reply:
I wondered about that too and signed overflow is indeed UB.
Converting it to unsigned (while keeping compatibility with the negative seeds) would be a bother...